kube-client/src/client/mod.rs

```diff
@@ -75,8 +75,8 @@ pub use builder::{ClientBuilder, DynBody};
 #[derive(Clone)]
 pub struct Client {
     // - `Buffer` for cheap clone
-    // - `BoxService` for dynamic response future type
-    inner: Buffer<BoxService<Request<Body>, Response<Body>, BoxError>, Request<Body>>,
+    // - `BoxFuture` for dynamic response future type
+    inner: Buffer<Request<Body>, BoxFuture<'static, Result<Response<Body>, BoxError>>>,
```

Comment:
This does seem like a nicer signature split. Interesting that it has to be 'static though. Is that a problem you think?

Reply:
`BoxService`can only be created by moving the underlying Service - https://docs.rs/tower/latest/tower/util/struct.BoxService.html#impl-BoxService%3CT,+U,+E%3E whereas a `BoxFuture` can be created from a [ref](https://docs.rs/futures/latest/futures/future/type.BoxFuture.html#impl-Pin%3CP%3E-1), hence generic over lifetime.

The `kube::Client` design owns the Buffer and to make Buffer `Clone` the Future needs to be `'static` - <https://tower-rs.github.io/tower/tower/buffer/struct.Buffer.html#impl-Clone-for-Buffer%3CReq,+F%3E>

So `'static` here is both unavoidable & fine (IMO!)

Reply:
Similar in the Tonic upgrade PR - https://github.com/hyperium/tonic/pull/1892/files

Reply:
Makes sense. Thanks for explaining!